tests/unit/heap.c

```diff
@@ -23,9 +23,66 @@
 
 #include <igraph.h>
 
+#include "test_utilities.inc"
+
 int main() {
 
-    /* This is not used by any functions any more,
-       no need to test it right now */
+    igraph_heap_t h;
+    igraph_integer_t i;
+    igraph_real_t *arr =(igraph_real_t*) calloc(5, sizeof(igraph_real_t));
+
+    /* igraph_heap_init, igraph_heap_destroy  */
+    printf("Test igraph_heap_init, igraph_heap_destroy\n");
+    igraph_heap_init(&h, 0);
+    igraph_heap_destroy(&h);
+    igraph_heap_init(&h, 10);
+    igraph_heap_destroy(&h);
+
+    /* igraph_heap_reserve */
+    printf("Test igraph_heap_reserve\n");
+    igraph_heap_init(&h, 0);
+    igraph_heap_reserve(&h, 10);
+    igraph_heap_reserve(&h, 5);
+    igraph_heap_destroy(&h);
+
+    /* igraph_heap_init_array*/
+    printf("Test igraph_heap_init_array\n");
+    igraph_heap_init_array(&h,arr,5);
+
+    /* igraph_heap_empty igraph_heap_size */
+    printf("Test igraph_heap_size, igraph_heap_empty\n");
+    IGRAPH_ASSERT((igraph_heap_size(&h)==5) && (!igraph_heap_empty(&h)));
+    igraph_heap_push(&h,100);
+    IGRAPH_ASSERT((igraph_heap_size(&h)==6) && (!igraph_heap_empty(&h)));
+    igraph_heap_delete_top(&h);
+    IGRAPH_ASSERT((igraph_heap_size(&h)==5) && (!igraph_heap_empty(&h)));
+    igraph_heap_destroy(&h);
+    igraph_heap_init(&h, 0);
+    IGRAPH_ASSERT((igraph_heap_size(&h)==0) && (igraph_heap_empty(&h)));
+
+    /* igraph_heap_push igraph_heap_top  */
+    printf("Test igraph_heap_push, igraph_heap_top\n");
+    igraph_real_t random_list[13]={-9.999,0,6,235,-2,-1000,-1,4,2000,0.1,1,-9,10};
+    igraph_vector_t top_sequence;
+    igraph_vector_init(&top_sequence,13);
+    for (i=0;i<13;i++)
+    {
+        igraph_heap_push(&h,random_list[i]);
+        VECTOR(top_sequence)[i]=igraph_heap_top(&h);
+    }
+    print_vector_format(&top_sequence,stdout,"%g");
```

Comment:
This is the same as `print_vector(&top_sequence)`. However, let us keep things simple (and the risk of future mistakes low) and print the heap elements directly from the `for` loop instead of writing them into a vector first.